sdk/servicebus/azure-messaging-servicebus/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,152 @@
+The default value of this property is `10 * number of CPU cores`. This problem is encountered more 
+frequently in an AKS environment.
+
+The reason this occurs is because of thread starvation. When all threads are utilized for message processing, the 
+processing thread can't get new threads to run other tasks. And the reason for thread starvation are two-fold. 
+One of them is the backpressure from the reactor-operators-chain which causes the acknowledgements to stay in the
+receiveLinkHandler's buffer. The other is the problem with credit calculation. More information on that is below.
+#### Credit calculation issue
+Currently, the credits that are placed on the link are overestimated as the way we calculated the number of credits 
+used is incorrect. Trying to get link Credits from multiple threads to calculate credits fail. This is something we 
+are actively working on. This section will be updated in the future when there is a resolution (most likely upgrading 
+to the version including the fix).
+
+## Autocomplete issue
+The way autocomplete and auto lock renewal works today in the `ServiceBusReceiverAsyncClient` and 
+`ServiceBusProcessorClient` is incorrect. It makes an assumption that there are no buffers / prefetch queues on the 
+customer side.
+
+This problem also manifests itself due to thread hopping in the reactor operators, which might cause the completion to
+occur out of order, resulting in incorrect message state. 
+
+Mitigation : Use `disableAutoComplete()` and `.maxAutoLockRenewalDuration(Duration.ZERO)` to turn off the two features 
+in `ServiceBusReceiverAsyncClient`. 
+Especially when there is any kind of buffering involved in the message processing code path. After disabling 
+AutoComplete, message settlement (completion / abandonment) should be done explicitly from the message processing code.
+For more information on how to use this correctly, please refer to the [samples][Samples]
+
+## Migrate from legacy to new client library
+The [migration guide][MigrationGuide] includes steps on migrating from the legacy client and migrating legacy 
+checkpoints.
+
+## Enable and configure logging
+The Azure SDK for Java offers a consistent logging story to help troubleshoot application errors and expedite their 
+resolution. The logs produced will capture the flow of an application before reaching the terminal state to help locate 
+the root issue.  View the [logging][Logging] wiki for guidance about enabling logging.
+
+In addition to enabling logging, setting the log level to `VERBOSE` or `DEBUG` provides insights into the library's 
+state.  Below are sample log4j2 and logback configurations to reduce the excessive messages when verbose logging is 
+enabled.
+
+### Configuring Log4J 2
+
+1. Add the dependencies in your pom.xml using ones from the [logging sample pom.xml][LoggingPom] under the "Dependencies required for Log4j2" section.
```

Comment:
nit: consistent spacing between headers and paragraphs. There is an empty line here and below, but the ones above do not have it.